r/tests/testthat/helper-expectation.R

```diff
@@ -98,12 +103,16 @@ expect_dplyr_equal <- function(expr, # A dplyr pipeline with `input` as its star
 expect_dplyr_error <- function(expr, # A dplyr pipeline with `input` as its start
```

Comment:
I did a bunch of digging through this + through git history to figure out what is going on here, and `expect_dplyr_error()` has a few issues on top of the one that you've caught here. This setup is super confusing since both the helper and the tests were wrong in different ways, making debugging this super hard (congrats on getting as much as you did here on your first PR, I wouldn't have gotten close to this!)

Ultimately what we want is for `expect_dplyr_error()` to evaluate the expression both as a standard R / dplyr based expression and then also as an arrow-backed dplyr expression and then we compare that the errors we get from the pure-dplyr version are the same as the errors we get from the arrow version.
 
 _it turns out_ that the way that dplyr errors / the way that we catch them here isn't quite right and so the tests below will fail.
 
 I resurrected the code I suggest below from git history + some gsubing on the errors. The crux of the issue so far is:

dplyr returns on error with the form:
```
Problem with `filter()` input `..1`.
x object 'b_var' not found
ℹ Input `..1` is `chr == b_var`.
```

but what we really care about is just the `x` block:
`object 'b_var' not found`

Which is all that arrow returns, but that should be good enough for us to consider them having the same error behavior.

The code I added below is probably not the right way to do this (since we would only be able to test a handful of error types this way) but I wanted to confirm that was what was going on.